diff -Nurw scintilla-scite/gtk/SciTEGTK.cxx scite/gtk/SciTEGTK.cxx
--- scintilla-scite/gtk/SciTEGTK.cxx	2017-05-26 00:25:02.000000000 +0200
+++ scite/gtk/SciTEGTK.cxx	2017-11-08 02:53:22.000000000 +0100
@@ -777,6 +777,11 @@
 	propsPlatform.Set("PLAT_GTK", "1");
 	propsPlatform.Set("PLAT_UNIX", "1");
 
+	// Make UNIX %HOME% available to scite config.	
+	FilePath envHome =getenv("HOME");
+	if (envHome.IsDirectory())
+		propsPlatform.Set("env.home", envHome.AsUTF8().c_str());
+
 	ReadEnvironment();
 
 	pathAbbreviations = GetAbbrevPropertiesFileName();
@@ -877,56 +882,69 @@
 	gtk_widget_show(button);
 	return button;
 }
+FilePath SciTEGTK::GetSciteDefaultHome() {
+	FilePath homePath;
 
-FilePath SciTEGTK::GetDefaultDirectory() {
-	const char *where = getenv("SciTE_HOME");
-#ifdef SYSCONF_PATH
-	if (!where) {
-		where = SYSCONF_PATH;
-	}
+#ifdef SYSCONF_PATH // default guaranteed to exist by OS
+ 	const std::string cdefault = SYSCONF_PATH; 
 #else
-	if (!where) {
-		where = getenv("HOME");
-	}
+	const std::string cdefault = getenv("HOME");
 #endif
-	if (where) {
-		return FilePath(where);
-	}
 
-	return FilePath("");
-}
+	// use plats SciTE_HOME
+	homePath=getenv("SciTE_HOME");
+	if (homePath.Exists())
+		return homePath;
 
-FilePath SciTEGTK::GetSciteDefaultHome() {
-	const char *where = getenv("SciTE_HOME");
-#ifdef SYSCONF_PATH
-	if (!where) {
-		where = SYSCONF_PATH;
-	}
-#else
-	if (!where) {
-		where = getenv("HOME");
-	}
-#endif
-	if (where) {
-		return FilePath(where);
+	homePath = SciTEGTK::GetDefaultDirectory();
+	if (homePath.Exists())
+		return homePath;
 
-	}
-	return FilePath("");
+	return FilePath(cdefault);
 }
 
 FilePath SciTEGTK::GetSciteUserHome() {
-	// First looking for environment variable $SciTE_USERHOME
-	// to set SciteUserHome. If not present we look for $SciTE_HOME
-	// then defaulting to $HOME
-	char *where = getenv("SciTE_USERHOME");
-	if (!where) {
-		where = getenv("SciTE_HOME");
-		if (!where) {
-			where = getenv("HOME");
+/** First looking for environment variable $SciTE_USERHOME
+* to set SciteUserHome. If not present we use GetSciteDefaultHome
+*/
+
+	// 1 set & use scite_home from scites env.scite_userhome
+	std::string home=props.GetNewExpandString("env.scite_userhome");
+	home=FilePath(home).NormalizePath().AsUTF8().c_str();	
+	if (home.find("/") != std::string::npos) {
+		putenv( (char *) ("SciTE_USERHOME="+home).c_str() );	
+		return FilePath(home);
+	}
+	
+	// try plats scite_userhome
+	FilePath homePath=getenv("SciTE_USERHOME");
+        homePath.NormalizePath();
+		if (homePath.Exists())
+			return homePath;
+
+	// use fallback, guranteed to exist by OS.
+    home=SciTEGTK::GetSciteDefaultHome().AsUTF8().c_str(); 	
+    putenv( (char *) ("SciTE_USERHOME="+home).c_str() );
+    
+	return FilePath(home);
 		}
+
+FilePath SciTEGTK::GetDefaultDirectory() {
+	// We are using executables binPath
+	std::string envHome;
+	char buf[PATH_MAX + 1];	
+	
+	if (readlink("/proc/self/exe", buf, sizeof(buf) - 1) >0) {	
+		envHome = buf;         
+	} else {
+		// Dont force proc to be available.	
+		envHome = sciteExecutable.AsInternal();
 	}
 
-	return FilePath(where);
+    // strip down to last path
+	    envHome = envHome.substr(0, envHome.rfind(pathSepChar));
+
+	return  FilePath(envHome);
 }
 
 void SciTEGTK::SetStatusBarText(const char *s) {
diff -Nurw scintilla-scite/win32/SciTEWin.cxx scite/win32/SciTEWin.cxx
--- scintilla-scite/win32/SciTEWin.cxx	2017-05-26 00:25:04.000000000 +0200
+++ scite/win32/SciTEWin.cxx	2017-11-08 02:50:34.000000000 +0100
@@ -208,6 +208,11 @@
 	propsPlatform.Set("PLAT_WIN", "1");
 	propsPlatform.Set("PLAT_WINNT", "1");
 
+	// Make Windows %USERPROFILE% available to scite config.
+	FilePath envHome =_wgetenv(GUI_TEXT("USERPROFILE"));
+	if (envHome.IsDirectory()) 
+	propsPlatform.Set("env.home", envHome.AsUTF8().c_str());
+	
 	ReadEnvironment();
 
 	ReadGlobalPropFile();
@@ -451,43 +456,61 @@
 	}
 }
 
-static FilePath GetSciTEPath(const FilePath &home) {
-	if (home.IsSet()) {
-		return FilePath(home);
-	} else {
-		GUI::gui_char path[MAX_PATH];
-		if (::GetModuleFileNameW(0, path, ELEMENTS(path)) == 0)
-			return FilePath();
-		// Remove the SciTE.exe
-		GUI::gui_char *lastSlash = wcsrchr(path, pathSepChar);
-		if (lastSlash)
-			*lastSlash = '\0';
-		return FilePath(path);
-	}
-}
-
-FilePath SciTEWin::GetDefaultDirectory() {
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-	return GetSciTEPath(home);
-}
 
 FilePath SciTEWin::GetSciteDefaultHome() {
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-	return GetSciTEPath(home);
+/**
+ *		SciteDefaultHome -> Windows
+ *		- we return GetDefaultDirectory		
+ */
+
+	return(SciTEWin::GetDefaultDirectory());
 }
 
 FilePath SciTEWin::GetSciteUserHome() {
-	// First looking for environment variable $SciTE_USERHOME
-	// to set SciteUserHome. If not present we look for $SciTE_HOME
-	// then defaulting to $USERPROFILE
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_USERHOME"));
-	if (!home) {
-		home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-		if (!home) {
-			home = _wgetenv(GUI_TEXT("USERPROFILE"));
+/**
+ *		SciteUserHome -> Windows
+ *		1. We look for and follow $(env.scite_userhome) or %SciTE_USERHOME% 
+ *		2. Else, we use GetSciteDefaultHome 
+ */
+
+	// First, check if  SciTE_UserHome has been set via property.
+	std::wstring wenvPathSciteHome = (GUI::StringFromUTF8(props.GetNewExpandString("env.scite_userhome")));
+	std::wstring wenv = GUI::StringFromUTF8(FilePath(L"SciTE_USERHOME=" + wenvPathSciteHome).NormalizePath().AsUTF8());
+	if (!wenvPathSciteHome.empty()) {
+			_wputenv((wchar_t *)wenv.c_str()); 
+			return(FilePath((wchar_t *)wenvPathSciteHome.c_str()));
 		}
+	
+	// No, lets  look for preset environment variable $SciTE_USERHOME. 
+	FilePath fpUserHome = _wgetenv(GUI_TEXT("SciTE_USERHOME"));
+	if (fpUserHome.Exists())
+		return(fpUserHome);
+	
+	// Fallback - just returnGetSciteDefaultHome()	
+	return SciTEWin::GetSciteDefaultHome();			
 	}
-	return GetSciTEPath(home);
+
+FilePath SciTEWin::GetDefaultDirectory() {
+/**
+ *		SciteDefaultDirectory -> Windows
+ *		1. We look for and follow %SciTE_HOME% 
+ *		2. Or we use exectables Path.
+ */
+
+	FilePath envHome =_wgetenv(GUI_TEXT("SciTE_HOME"));
+	if (envHome.IsDirectory()) 
+		return envHome;
+		
+	//  ..just use executables binpath
+		std::wstring wPath;
+		GUI::gui_char path[MAX_PATH];
+		if (::GetModuleFileNameW(0, path, ELEMENTS(path)) != 0) {
+			GUI::gui_char *lastSlash = wcsrchr(path, pathSepChar);
+			if (lastSlash) *lastSlash = '\0';
+				envHome = path;
+	}
+
+	return envHome;
 }
 
 // Help command lines contain topic!path
