diff -Nurw scintilla-scite/src/PropSetFile.cxx scite/src/PropSetFile.cxx
--- scintilla-scite/src/PropSetFile.cxx	2017-05-26 00:25:04.000000000 +0200
+++ scite/src/PropSetFile.cxx	2017-11-08 03:45:32.000000000 +0100
@@ -424,6 +424,9 @@
 				}
 			}
 		} else if (filter.IsValid(importName)) {
+				importName = Expand(importName.c_str()); /// Allow using $() Variables
+				if (importName[0]==pathSepChar)
+				importName=importName.substr(1,std::string::npos);
 			importName += ".properties";
 			FilePath importPath(directoryForImports, FilePath(GUI::StringFromUTF8(importName)));
 			Import(importPath, directoryForImports, filter, imports, depth+1);
