diff -Nurw scintilla-scite/gtk/SciTEGTK.cxx scite/gtk/SciTEGTK.cxx
--- scintilla-scite/gtk/SciTEGTK.cxx	2016-11-25 19:41:00.000000000 +0100
+++ scite/gtk/SciTEGTK.cxx	2016-11-29 20:04:04.000000000 +0100
@@ -2,6 +2,10 @@
 // SciTEGTK.cxx - main code for the GTK+ version of the editor
 // Copyright 1998-2004 by Neil Hodgson <neilh@scintilla.org>
 // The License.txt file describes the conditions under which this software may be distributed.
+/// 11.11.2016: add env.scite_userhome & env.home (T. Kani, Marcedo{at}habMalneFrage.de) 
+
+#include <unistd.h>
+#include <limits.h>
 
 #include <stdlib.h>
 #include <string.h>
@@ -15,6 +19,7 @@
 
 #include <string>
 #include <vector>
+#include <iostream>
 #include <set>
 #include <map>
 #include <algorithm>
@@ -775,6 +785,10 @@
 	timerID = 0;
 
 	PropSetFile::SetCaseSensitiveFilenames(true);
+	// System type properties are stored in the platform properties.
+	FilePath envHome =getenv("HOME");
+	if (envHome.IsDirectory())
+		propsPlatform.Set("env.home", envHome.AsUTF8().c_str())		
 	propsPlatform.Set("PLAT_GTK", "1");
 	propsPlatform.Set("PLAT_UNIX", "1");
 
@@ -880,55 +894,63 @@
 	return button;
 }
 
-FilePath SciTEGTK::GetDefaultDirectory() {
-	const char *where = getenv("SciTE_HOME");
-
-#ifdef SYSCONF_PATH
-	if (!where)
-		where = SYSCONF_PATH;
-
-#else
-	if (!where)
-		where = getenv("HOME");
-
-#endif
-	if (where)
-		return FilePath(where);
-
-	return FilePath("");
-}
-
 FilePath SciTEGTK::GetSciteDefaultHome() {
-	const char *where = getenv("SciTE_HOME");
-
-#ifdef SYSCONF_PATH
-	if (!where)
-		where = SYSCONF_PATH;
-
-#else
-	if (!where)
-		where = getenv("HOME");
-
-#endif
-	if (where)
-		return FilePath(where);
+	const std::string cdefault = "/usr/share/scite"; // default guaranteed to exist by OS
+	std::string home;
+	FilePath homePath;
+	
+	// 1 use SciTE_HOME
+	std::string envhome = getenv("SciTE_HOME");
+	homePath=envhome;
+	if (homePath.Exists())
+		return homePath;
+		
+	// 2 Search config in ~/scite
+	envhome = getenv("HOME");
+	homePath=envhome + "/scite/SciTEGlobal.properties";
+	if (homePath.Exists()) 
+		return FilePath(envhome+"/scite");
+		
+	// 3 Search confug in executables binPath
+	char buf[PATH_MAX + 1];
+	if (readlink("/proc/self/exe", buf, sizeof(buf) - 1) >0) {
+	// just get the path	
+		envhome = buf;
+		envhome = envhome.substr(0, envhome.rfind('/'));
+		homePath = envhome +"/SciTEGlobal.properties";
+		if (homePath.Exists())
+			return FilePath(envhome);	
+}
 
-	return FilePath("");
+	return FilePath(cdefault);
 }
 
 FilePath SciTEGTK::GetSciteUserHome() {
-	// First looking for environment variable $SciTE_USERHOME
-	// to set SciteUserHome. If not present we look for $SciTE_HOME
-	// then defaulting to $HOME
-	char *where = getenv("SciTE_USERHOME");
-
-	if (!where) {
-		where = getenv("SciTE_HOME");
-		if (!where)
-			where = getenv("HOME");
+/** First looking for environment variable $SciTE_USERHOME
+* to set SciteUserHome. If not present we look for $SciTE_HOME
+* then defaulting to $HOME
+*/
+
+	// 1 set & use scite_home from env.scite_home
+	std::string home=props.GetNewExpandString("env.scite_userhome");
+	home=FilePath(home).NormalizePath().AsUTF8().c_str();	
+	if (home.find("/") != std::string::npos) {
+		putenv( (char *) ("SciTE_USERHOME="+home).c_str() );	
+		return FilePath(home);
+	}
+	
+	//  try $scite_userhome
+	FilePath homePath=getenv("SciTE_USERHOME");
+	home=homePath.AsUTF8().c_str();
+	if (homePath.Exists())
+		return homePath;
+
+	// use fallback, guranteed to exist by OS.	 		 
+	return SciTEGTK::GetSciteDefaultHome();
 	}
 
-	return FilePath(where);
+FilePath SciTEGTK::GetDefaultDirectory() {
+	return SciTEGTK::GetSciteDefaultHome();
 }
 
 void SciTEGTK::ShowFileInStatus() {
diff -Nurw scintilla-scite/win32/SciTEWin.cxx scite/win32/SciTEWin.cxx
--- scintilla-scite/win32/SciTEWin.cxx	2016-09-04 02:17:28.000000000 +0200
+++ scite/win32/SciTEWin.cxx	2016-11-29 20:02:50.000000000 +0100
@@ -4,6 +4,7 @@
  **/
 // Copyright 1998-2003 by Neil Hodgson <neilh@scintilla.org>
 // The License.txt file describes the conditions under which this software may be distributed.
+/// 11.11.2016: add env.scite_userhome & env.home (T. Kani, Marcedo{at}habMalneFrage.de) 
 
 #include <time.h>
 
@@ -175,10 +176,13 @@
 	subProcessGroupId = 0;
 
 	pathAbbreviations = GetAbbrevPropertiesFileName();
-
 	// System type properties are stored in the platform properties.
 	propsPlatform.Set("PLAT_WIN", "1");
 	propsPlatform.Set("PLAT_WINNT", "1");
+	FilePath envHome =_wgetenv(GUI_TEXT("USERPROFILE"));
+	if (envHome.IsDirectory()) 
+	propsPlatform.Set("env.home", envHome.AsUTF8().c_str());
 
 	ReadEnvironment();
 
@@ -417,43 +422,74 @@
 	SciTEBase::ReadProperties();
 }
 
-static FilePath GetSciTEPath(FilePath home) {
-	if (home.IsSet()) {
-		return FilePath(home);
-	} else {
+
+FilePath SciTEWin::GetSciteDefaultHome() {
+/**
+ *       SciteDefaultHome -> Windows
+ *       1 look for and follow %SciTE_HOME% | $(env.scite_home)
+ *       2 else use exectables Path (if we find SciteGlobal.properties)
+ *       3 else use %USERPROFILE%\scite\ (if we find SciteGlobal.properties there)
+ */
+				
+	//  ..try SciTE_HOME
+	FilePath envHome =_wgetenv(GUI_TEXT("SciTE_HOME"));
+	if (envHome.IsDirectory())
+		return envHome;
+		
+	std::wstring home;
+		
+	//  ..try executables binpath (when we find sciteglobal.properties there.)
+	if (home.empty()) {
+		std::wstring wPath;
 		GUI::gui_char path[MAX_PATH];
-		if (::GetModuleFileNameW(0, path, ELEMENTS(path)) == 0)
-			return FilePath();
-		// Remove the SciTE.exe
+		if (::GetModuleFileNameW(0, path, ELEMENTS(path)) != 0) {
+			//  just get  the Path
 		GUI::gui_char *lastSlash = wcsrchr(path, pathSepChar);
-		if (lastSlash)
-			*lastSlash = '\0';
+			if (lastSlash) *lastSlash = '\0';
+				wPath = path;
+			FilePath wfilePath = wPath.append(L"\\SciTEGlobal.properties");
+			if (wfilePath.Exists())
 		return FilePath(path);
 	}
 }
 
-FilePath SciTEWin::GetDefaultDirectory() {
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-	return GetSciTEPath(home);
+	// yo.... FilePath takes and returns gui_string (which is a basic_wstring / wchar_t)
+	// To get a std::wstring back use GUI::UTF8FromString(FilePath(xyz)).ToUTF8();
+
+	// if above are empty... define folder %userprofile%\scite as a fallback.
+	if (home.empty()) {
+		std::wstring wPath = _wgetenv(GUI_TEXT("USERPROFILE"));
+		FilePath wfilePath = FilePath(wPath + L"\\scite" ).NormalizePath();
+		if (wfilePath.IsDirectory())
+			home = wPath;
 }
 
-FilePath SciTEWin::GetSciteDefaultHome() {
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-	return GetSciTEPath(home);
+	return(home);
 }
 
 FilePath SciTEWin::GetSciteUserHome() {
+	GUI::gui_char *home ;
+	
+	// Set environment %SciTE_HOME% fromm $(env.scite_userhome).
+	std::wstring wenvPathSciteHome = (GUI::StringFromUTF8(props.GetNewExpandString("env.scite_userhome")));
+	std::wstring wenv = GUI::StringFromUTF8(FilePath(L"SciTE_HOME=" + wenvPathSciteHome).NormalizePath().AsUTF8());
+	if (wenv.find(L"\\") != std::string::npos) {
+		_wputenv((wchar_t *)wenv.c_str()); 
+		return(FilePath((wchar_t *)wenvPathSciteHome.c_str()));
+		}
+		
 	// First looking for environment variable $SciTE_USERHOME
 	// to set SciteUserHome. If not present we look for $SciTE_HOME
 	// then defaulting to $USERPROFILE
-	GUI::gui_char *home = _wgetenv(GUI_TEXT("SciTE_USERHOME"));
-	if (!home) {
-		home = _wgetenv(GUI_TEXT("SciTE_HOME"));
-		if (!home) {
-			home = _wgetenv(GUI_TEXT("USERPROFILE"));
-		}
+	home = _wgetenv(GUI_TEXT("SciTE_USERHOME"));
+	if (home)
+		return(FilePath(home));
+		 	
+	return SciTEWin::GetSciteDefaultHome();
 	}
-	return GetSciTEPath(home);
+
+FilePath SciTEWin::GetDefaultDirectory() {
+	return SciTEWin::GetSciteDefaultHome();
 }
 
 // Help command lines contain topic!path
