diff -Nurw scintilla-scite/src/SciTE.h scite/src/SciTE.h
--- scintilla-scite/src/SciTE.h	2016-10-08 14:54:00.000000000 +0200
+++ scite/src/SciTE.h	2016-10-08 14:55:28.000000000 +0200
@@ -53,7 +53,7 @@
 #define IDM_ENCODING_UCOOKIE	154
 
 #define MRU_START	17
-#define IMPORT_START 20
+#define IMPORT_START 8 // Itemposition to add config filenames (Menu-Options-Config Files)
 #define TOOLS_START 4
 
 // Edit
diff -Nurw scintilla-scite/src/SciTEBase.cxx scite/src/SciTEBase.cxx
--- scintilla-scite/src/SciTEBase.cxx	2016-09-04 02:17:28.000000000 +0200
+++ scite/src/SciTEBase.cxx	2016-10-08 14:55:28.000000000 +0200
@@ -4218,18 +4218,18 @@
 	props.Set("SciteUserHome", homepath.AsUTF8().c_str());
 }
 
+
 void SciTEBase::UIAvailable() {
-	SetImportMenu();
+ // Decide if we want to display language files (xxx.properties) within the options Menu 
+ SetImportMenu(props.GetInt("menu.options.showUserProps"));
 	if (extender) {
-		SetHomeProperties();
+		SetHomeProperties(); // Define SciteDefault- and SciteUserHome 
 		extender->Initialise(this);
 	}
 }
 
-/**
- * Find the character following a name which is made up of characters from
- * the set [a-zA-Z.]
- */
+// Find the character following a name which is made up of characters from the set [a-zA-Z.]
+
 static GUI::gui_char AfterName(const GUI::gui_char *s) {
 	while (*s && ((*s == '.') ||
 	        (*s >= 'a' && *s <= 'z') ||
diff -Nurw scintilla-scite/src/SciTEBase.h scite/src/SciTEBase.h
--- scintilla-scite/src/SciTEBase.h	2016-10-08 14:54:00.000000000 +0200
+++ scite/src/SciTEBase.h	2016-10-08 14:55:28.000000000 +0200
@@ -821,10 +821,16 @@
 	virtual void SizeContentWindows() = 0;
 	virtual void SizeSubWindows() = 0;
 
+  // this one can also add Menuentries to a Submenu popup 
+  virtual void SetMenuItemNew (int menuNumber, int subMenuNumber, int position, int itemID,
+		const GUI::gui_char *text, const GUI::gui_char *mnemonic = 0) = 0;
+  // this one will be migrated.  
 	virtual void SetMenuItem(int menuNumber, int position, int itemID,
 		const GUI::gui_char *text, const GUI::gui_char *mnemonic = 0) = 0;
+    
 	virtual void RedrawMenu() {}
 	virtual void DestroyMenuItem(int menuNumber, int itemID) = 0;
+	
 	virtual void CheckAMenuItem(int wIDCheckItem, bool val) = 0;
 	virtual void EnableAMenuItem(int wIDCheckItem, bool val) = 0;
 	virtual void CheckMenusClipboard();
@@ -854,7 +860,7 @@
 	void AssignKey(int key, int mods, int cmd);
 	void ViewWhitespace(bool view);
 	void SetAboutMessage(GUI::ScintillaWindow &wsci, const char *appTitle);
-	void SetImportMenu();
+	void SetImportMenu(int displayPropFiles);
 	void ImportMenu(int pos);
 	void SetLanguageMenu();
 	void SetPropertiesInitial();
diff -Nurw scintilla-scite/src/SciTEIO.cxx scite/src/SciTEIO.cxx
--- scintilla-scite/src/SciTEIO.cxx	2016-09-04 02:17:28.000000000 +0200
+++ scite/src/SciTEIO.cxx	2016-10-08 14:55:28.000000000 +0200
@@ -1005,10 +1005,10 @@
 
 void SciTEBase::ReloadProperties() {
 	ReadGlobalPropFile();
-	SetImportMenu();
 	ReadLocalPropFile();
 	ReadAbbrevPropFile();
 	ReadProperties();
+  SetImportMenu(props.GetInt("menu.options.showUserProps"));
 	SetWindowName();
 	BuffersMenu();
 	Redraw();
diff -Nurw scintilla-scite/src/SciTEProps.cxx scite/src/SciTEProps.cxx
--- scintilla-scite/src/SciTEProps.cxx	2016-10-08 14:54:00.000000000 +0200
+++ scite/src/SciTEProps.cxx	2016-10-08 14:55:28.000000000 +0200
@@ -52,10 +52,12 @@
 #include "MatchMarker.h"
 #include "SciTEBase.h"
 
-void SciTEBase::SetImportMenu() {
+void SciTEBase::SetImportMenu(int iShow) {
+// Extends Options Menu
 	for (int i = 0; i < importMax; i++) {
-		DestroyMenuItem(menuOptions, importCmdID + i);
+     if (iShow==0) DestroyMenuItem(menuOptions,1300+ i);
 	}
+  //importCmdID
 	if (!importFiles.empty()) {
 		for (int stackPos = 0; stackPos < static_cast<int>(importFiles.size()) && stackPos < importMax; stackPos++) {
 			int itemID = importCmdID + stackPos;
@@ -63,12 +65,13 @@
 				GUI::gui_string entry = localiser.Text("Open");
 				entry += GUI_TEXT(" ");
 				entry += importFiles[stackPos].Name().AsInternal();
-				SetMenuItem(menuOptions, IMPORT_START + stackPos, itemID, entry.c_str());
+        if (iShow==1) SetMenuItemNew(menuOptions, 9, IMPORT_START+stackPos, itemID, entry.c_str());
 			}
 		}
 	}
 }
 
+
 void SciTEBase::ImportMenu(int pos) {
 	if (pos >= 0) {
 		if (importFiles[pos].IsSet()) {
@@ -78,6 +81,7 @@
 }
 
 void SciTEBase::SetLanguageMenu() {
+// Fills Language Menu
 	for (int i = 0; i < 100; i++) {
 		DestroyMenuItem(menuLanguage, languageCmdID + i);
 	}
diff -Nurw scintilla-scite/win32/SciTERes.rc scite/win32/SciTERes.rc
--- scintilla-scite/win32/SciTERes.rc	2016-10-08 14:54:00.000000000 +0200
+++ scite/win32/SciTERes.rc	2016-10-08 14:55:30.000000000 +0200
@@ -143,6 +143,8 @@
 	MENUITEM "Wrap Out&put",			IDM_WRAPOUTPUT
 	MENUITEM "&Read-Only",				IDM_READONLY
 	MENUITEM SEPARATOR
+	POPUP "Editor"
+  BEGIN
 	POPUP "&Line End Characters"
 	BEGIN
 		MENUITEM "CR &+ LF",	IDM_EOL_CRLF
@@ -153,15 +155,19 @@
 	MENUITEM SEPARATOR
 	MENUITEM "Change Inden&tation Settings...\tCtrl+Shift+I",	IDM_TABSIZE
 	MENUITEM "Use &Monospaced Font\tCtrl+F11",	IDM_MONOFONT
+	END
 	MENUITEM SEPARATOR
-	MENUITEM "Open Local &Options File",		IDM_OPENLOCALPROPERTIES
-	MENUITEM "Open &Directory Options File",		IDM_OPENDIRECTORYPROPERTIES
+  POPUP "Config Files"
+  BEGIN
+//    MENUITEM "Open Local Options File",		IDM_OPENLOCALPROPERTIES
+//    MENUITEM "Open Directory Options File",		IDM_OPENDIRECTORYPROPERTIES
 	MENUITEM "Open &User Options File",		IDM_OPENUSERPROPERTIES
 	MENUITEM "Open &Global Options File",		IDM_OPENGLOBALPROPERTIES
 	MENUITEM "Open A&bbreviations File",		IDM_OPENABBREVPROPERTIES
-	MENUITEM "Open Lua Startup Scr&ipt", 		IDM_OPENLUAEXTERNALFILE
+//	MENUITEM "Open Lua Startup Scr&ipt", 		IDM_OPENLUAEXTERNALFILE
+	END
 	MENUITEM SEPARATOR
-END // used: abcghlmoprstux
+END // used: abcghilmprstux
 
 POPUP "&Language"
 BEGIN
diff -Nurw scintilla-scite/win32/SciTEWin.h scite/win32/SciTEWin.h
--- scintilla-scite/win32/SciTEWin.h	2016-10-08 14:54:00.000000000 +0200
+++ scite/win32/SciTEWin.h	2016-10-08 14:55:28.000000000 +0200
@@ -253,6 +253,10 @@
 	virtual void SizeContentWindows();
 	virtual void SizeSubWindows();
 
+  // Supports creation of Entries in SubMenus too
+  virtual void SetMenuItemNew(int menuNumber, int subMenuNumber, int position, int itemID,
+                           const GUI::gui_char *text, const GUI::gui_char *mnemonic =0); 
+
 	virtual void SetMenuItem(int menuNumber, int position, int itemID,
 	                         const GUI::gui_char *text, const GUI::gui_char *mnemonic = 0);
 	virtual void RedrawMenu();
diff -Nurw scintilla-scite/win32/SciTEWinBar.cxx scite/win32/SciTEWinBar.cxx
--- scintilla-scite/win32/SciTEWinBar.cxx	2016-10-08 14:54:00.000000000 +0200
+++ scite/win32/SciTEWinBar.cxx	2016-10-08 14:55:30.000000000 +0200
@@ -632,6 +632,47 @@
 	CheckMenus();
 }
 
+void SciTEWin::SetMenuItemNew(int menuNumber, int subMenuNumber, int position, int itemID,
+                           const GUI::gui_char *text, const GUI::gui_char *mnemonic) {
+	// On Windows the menu items are modified if they already exist or are created
+    HMENU hmenu = ::GetSubMenu(::GetMenu(MainHWND()), menuNumber);
+ // About to modify a submenu (eg Options-> Config Files)   
+	if(subMenuNumber >0) {
+    HMENU smenu = ::GetSubMenu(hmenu, subMenuNumber);
+    hmenu = smenu;
+  }
+	GUI::gui_string sTextMnemonic = text;
+	long keycode = 0;
+	if (mnemonic && *mnemonic) {
+		keycode = SciTEKeys::ParseKeyCode(GUI::UTF8FromString(mnemonic).c_str());
+		if (keycode) {
+			sTextMnemonic += GUI_TEXT("\t");
+			sTextMnemonic += mnemonic;
+		}
+		// the keycode could be used to make a custom accelerator table
+		// but for now, the menu's item data is used instead for command
+		// tools, and for other menu entries it is just discarded.
+	}
+  
+	UINT typeFlags = (text[0]) ? MF_STRING : MF_SEPARATOR;
+	if (::GetMenuState(hmenu, itemID, MF_BYCOMMAND) == (UINT)(-1)) {
+		// Not present so insert
+		::InsertMenuW(hmenu, position, MF_BYPOSITION | typeFlags, itemID, sTextMnemonic.c_str());
+	} else {
+		::ModifyMenuW(hmenu, itemID, MF_BYCOMMAND | typeFlags, itemID, sTextMnemonic.c_str());
+	}
+
+	if (itemID >= IDM_TOOLS && itemID < IDM_TOOLS + toolMax) {
+		// Stow the keycode for later retrieval.
+		// Do this even if 0, in case the menu already existed (e.g. ModifyMenu)
+		MENUITEMINFO mii;
+		mii.cbSize = sizeof(MENUITEMINFO);
+		mii.fMask = MIIM_DATA;
+		mii.dwItemData = keycode;
+		::SetMenuItemInfo(hmenu, itemID, FALSE, &mii);
+	}
+}
+
 
 void SciTEWin::SetMenuItem(int menuNumber, int position, int itemID,
                            const GUI::gui_char *text, const GUI::gui_char *mnemonic) {
@@ -674,6 +715,7 @@
 	::DrawMenuBar(HwndOf(wSciTE));
 }
 
+
 void SciTEWin::DestroyMenuItem(int menuNumber, int itemID) {
 	// On Windows menu items are destroyed as they can not be hidden and they can be recreated in any position
 	HMENU hmenuBar = ::GetMenu(MainHWND());
diff -Nurw scintilla-scite/gtk/SciTEGTK.cxx scite/gtk/SciTEGTK.cxx
--- scintilla-scite/gtk/SciTEGTK.cxx	2016-11-19 17:57:44.000000000 +0100
+++ scite/gtk/SciTEGTK.cxx	2016-11-19 17:55:40.000000000 +0100
@@ -15,6 +15,8 @@
 #include <string>
 #include <vector>
+#include <iostream>
+#include <unistd.h>
 #include <set>
 #include <map>
 #include <algorithm>
  @@ -1343,6 +1361,49 @@
 	}
 }
 
+void SciTEGTK::SetMenuItemNew(int, int, int, int itemID, const char *text, const char *mnemonic) {
+	DestroyMenuItem(0, itemID);
+
+	// On GTK+ the menuNumber and position are ignored as the menu item already exists and is in the right
+	// place so only needs to be shown and have its text set.
+
+	std::string itemText = GtkFromWinCaption(text);
+
+	long keycode = 0;
+	if (mnemonic && *mnemonic) {
+		keycode = SciTEKeys::ParseKeyCode(mnemonic);
+		if (keycode) {
+			itemText += " ";
+			itemText += mnemonic;
+		}
+		// the keycode could be used to make a custom accelerator table
+		// but for now, the menu's item data is used instead for command
+		// tools, and for other menu entries it is just discarded.
+	}
+
+	// Reorder shift and ctrl indicators for compatibility with other menus
+	Substitute(itemText, "Ctrl+Shift+", "Shift+Ctrl+");
+
+	GtkWidget *item = MenuItemFromAction(itemID);
+	if (item) {
+		GList *al = gtk_container_get_children(GTK_CONTAINER(item));
+		for (unsigned int ii = 0; ii < g_list_length(al); ii++) {
+			gpointer d = g_list_nth(al, ii);
+			GtkWidget **w = (GtkWidget **)d;
+			gtk_label_set_text_with_mnemonic(GTK_LABEL(*w), itemText.c_str());
+		}
+		g_list_free(al);
+		gtk_widget_show(item);
+
+		if (itemID >= IDM_TOOLS && itemID < IDM_TOOLS + toolMax) {
+			// Stow the keycode for later retrieval.
+			// Do this even if 0, in case the menu already existed (e.g. ModifyMenu)
+			g_object_set_data(G_OBJECT(item), "key", GINT_TO_POINTER(static_cast<int>(keycode)));
+		}
+	}
+}
+
+
 void SciTEGTK::DestroyMenuItem(int, int itemID) {
 	// On GTK+ menu items are just hidden rather than destroyed as they can not be recreated in the middle of a menu
 	// The menuNumber is ignored as all menu items in GTK+ can be found from the root of the menu tree
@@ -3927,23 +3988,25 @@
 		{ "/Options/Wrap Out_put",		   "",			menuSig, IDM_WRAPOUTPUT,	      "<CheckItem>"			     },
 		{ "/Options/_Read-Only",		   "",			menuSig, IDM_READONLY,		      "<CheckItem>"			     },
 		{ "/Options/sep1",			   NULL,		NULL,	 0,			      "<Separator>"			     },
-		{ "/Options/_Line End Characters",	   "",			0,	 0,			      "<Branch>"			     },
-		{ "/Options/Line End Characters/CR _+ LF", "",			menuSig, IDM_EOL_CRLF,		      "<RadioItem>"			     },
-		{ "/Options/Line End Characters/_CR",	   "",			menuSig, IDM_EOL_CR,		      "/Options/Line End Characters/CR + LF" },
-		{ "/Options/Line End Characters/_LF",	   "",			menuSig, IDM_EOL_LF,		      "/Options/Line End Characters/CR + LF" },
-		{ "/Options/_Convert Line End Characters", "",			menuSig, IDM_EOL_CONVERT,	      0					     },
-		{ "/Options/sep2",			   NULL,		NULL,	 0,			      "<Separator>"			     },
-		{ "/Options/Change Inden_tation Settings", "<control><shift>I", menuSig, IDM_TABSIZE,		      0					     },
-		{ "/Options/Use _Monospaced Font",	   "<control>F11",	menuSig, IDM_MONOFONT,		      "<CheckItem>"			     },
+		{ "/Options/Editor/",				  "",		       0,	0,			     "<Branch>"					   },
+		{ "/Options/Editor/_Line End Characters",	  "",		       0,	0,			     "<Branch>"					   },
+		{ "/Options/Editor/Line End Characters/CR _+ LF", "",		       menuSig, IDM_EOL_CRLF,		     "<RadioItem>"				   },
+		{ "/Options/Editor/Line End Characters/_CR",	  "",		       menuSig, IDM_EOL_CR,		     "/Options/Editor/Line End Characters/CR + LF" },
+		{ "/Options/Editor/Line End Characters/_LF",	  "",		       menuSig, IDM_EOL_LF,		     "/Options/Editor/Line End Characters/CR + LF" },
+		{ "/Options/Editor/_Convert Line End Characters", "",		       menuSig, IDM_EOL_CONVERT,	     0						   },
+		{ "/Options/Editor/sep2",			  NULL,		       NULL,	0,			     "<Separator>"				   },
+		{ "/Options/Editor/Change Inden_tation Settings", "<control><shift>I", menuSig, IDM_TABSIZE,		     0						   },
+		{ "/Options/Editor/Use _Monospaced Font",	  "<control>F11",      menuSig, IDM_MONOFONT,		     "<CheckItem>"				   },
 		{ "/Options/sep3",			   NULL,		NULL,	 0,			      "<Separator>"			     },
-		{ "/Options/Open Local _Options File",	   "",			menuSig, IDM_OPENLOCALPROPERTIES,     0					     },
-		{ "/Options/Open _Directory Options File", "",			menuSig, IDM_OPENDIRECTORYPROPERTIES, 0					     },
-		{ "/Options/Open _User Options File",	   "",			menuSig, IDM_OPENUSERPROPERTIES,      0					     },
-		{ "/Options/Open _Global Options File",	   "",			menuSig, IDM_OPENGLOBALPROPERTIES,    0					     },
-		{ "/Options/Open A_bbreviations File",	   "",			menuSig, IDM_OPENABBREVPROPERTIES,    0					     },
-		{ "/Options/Open Lua Startup Scr_ipt",	   "",			menuSig, IDM_OPENLUAEXTERNALFILE,     0					     },
-		{ "/Options/sep4",			   NULL,		NULL,	 0,			      "<Separator>"			     },
-		{ "/Options/_Edit Properties",		   "",			0,	 0,			      "<Branch>"			     },
+		{ "/Options/Config/",				  "",		       0,	0,			     "<Branch>"					   },
+		{ "/Options/Config/Open Local _Options File",	  "",		       menuSig, IDM_OPENLOCALPROPERTIES,     0						   },
+		{ "/Options/Config/Open _Directory Options File", "",		       menuSig, IDM_OPENDIRECTORYPROPERTIES, 0						   },
+		{ "/Options/Config/Open _User Options File",	  "",		       menuSig, IDM_OPENUSERPROPERTIES,	     0						   },
+		{ "/Options/Config/Open _Global Options File",	  "",		       menuSig, IDM_OPENGLOBALPROPERTIES,    0						   },
+		{ "/Options/Config/Open A_bbreviations File",	  "",		       menuSig, IDM_OPENABBREVPROPERTIES,    0						   },
+		{ "/Options/Config/Open Lua Startup Scr_ipt",	  "",		       menuSig, IDM_OPENLUAEXTERNALFILE,     0						   },
+		{ "/Options/Config/sep4",			  NULL,		       NULL,	0,			     "<Separator>"				   },
+		{ "/Options/Config/Edit Properties",		  "",		       0,	0,			     "<Branch>"					   },
 	};
 
 	SciTEItemFactoryEntry menuItemsLanguage[] = {
@@ -3982,7 +4045,7 @@
 	CreateTranslatedMenu(ELEMENTS(menuItems), menuItems);
 
 	CreateTranslatedMenu(ELEMENTS(menuItemsOptions), menuItemsOptions,
-			     50, "/Options/Edit Properties/Props", 0, IDM_IMPORT, 0);
+			     50, "/Options/Config/Edit Properties/Props", 0, IDM_IMPORT, 0);
 	CreateTranslatedMenu(ELEMENTS(menuItemsLanguage), menuItemsLanguage,
 			     100, "/Language/Language", 0, IDM_LANGUAGE, 0);
 	if (props.GetInt("buffers") > 1)
