diff -Nurw a/SciTEBase.h b/SciTEBase.h
--- a/SciTEBase.h	2025-05-07 21:10:11.447429000 +0200
+++ b/SciTEBase.h	2025-05-08 13:26:14.317550500 +0200
@@ -674,7 +674,6 @@
 	FilePath GetLocalPropertiesFileName();
 	FilePath GetAbbrevPropertiesFileName();
 	void OpenProperties(int propsFile);
-	static int GetMenuCommandAsInt(std::string commandName);
 	virtual void Print(bool) {}
 	virtual void PrintSetup() {}
 	void UserStripShow(const char * /* description */) override {}
@@ -979,6 +978,7 @@
 	// WorkerListener
 	virtual void PostOnMainThread(int cmd, Worker *pWorker) override = 0;
 	virtual void WorkerCommand(int cmd, Worker *pWorker);
+	static int GetMenuCommandAsInt(std::string commandName);
 };
 
 int ControlIDOfCommand(unsigned long);
