diff -Nurw scite/src/SciTEProps.cxx scite_m/src/SciTEProps.cxx
--- scite/src/SciTEProps.cxx	2017-05-26 00:25:04.000000000 +0200
+++ scite_m/src/SciTEProps.cxx	2018-01-01 17:56:02.000000000 +0100
@@ -1366,6 +1379,13 @@
 		TimerEnd(timerAutoSave);
 	}
 
+	const int forceLexNullSize=props.GetInt("max.style.size",10000000);
+	if (LengthDocument()>forceLexNullSize){
+		wEditor.Call(SCI_SETLEXER, 1);
+		wEditor.Call(SCI_CLEARDOCUMENTSTYLE);
+		props.Set("Language", "");
+	}
+		
 	firstPropertiesRead = false;
 	needReadProperties = false;
 }
