diff -Nurw scite/src/SciTEProps.cxx scite_mod/src/SciTEProps.cxx
--- scite/src/SciTEProps.cxx	2017-05-26 00:25:03.000000000 +0200
+++ scite_mod/src/SciTEProps.cxx	2018-12-18 14:07:38.104258600 +0100
@@ -106,21 +106,24 @@
 const GUI::gui_char propLocalFileName[] = GUI_TEXT("SciTE.properties");
 const GUI::gui_char propDirectoryFileName[] = GUI_TEXT("SciTEDirectory.properties");
 
+/** 
+Push system env on propsPlatform. Format Key=value
+*/
 void SciTEBase::ReadEnvironment() {
 #if defined(__unix__)
 	extern char **environ;
-	char **e = environ;
+	char **pEnv= environ;
 #else
-	char **e = _environ;
+	char **pEnv= _environ;
 #endif
-	for (; e && *e; e++) {
+	for (; pEnv&& *pEnv; pEnv++) {
 		char key[1024];
-		char *k = *e;
-		char *v = strchr(k, '=');
-		if (v && (static_cast<size_t>(v - k) < sizeof(key))) {
-			memcpy(key, k, v - k);
-			key[v - k] = '\0';
-			propsPlatform.Set(key, v + 1);
+		char *pMe= *pEnv; // Varname's start position 
+		char *pValue = strchr(pMe, '='); // Values start position
+		if (pValue && ((int)(pValue - pMe) < (int)(sizeof(key)))) { // Validate length 
+			memcpy(key, pMe, pValue - pMe);
+			key[pValue - pMe] = '\0';
+			propsPlatform.Set(key, pValue + 1);
 		}
 	}
 }
