--- ScintillaBase.cxx	2017-12-04 16:29:50.238126900 +0100
+++ ScintillaBase_m.cxx	2017-12-04 16:29:54.238336900 +0100
@@ -254,8 +254,11 @@
 			return;
 		}
 	}
+	
+	int szEntry= (vs.styles[STYLE_DEFAULT].size / SC_FONT_SIZE_MULTIPLIER) +6;
+	
 	ac.Start(wMain, idAutoComplete, sel.MainCaret(), PointMainCaret(),
-				lenEntered, vs.lineHeight, IsUnicodeMode(), technology);
+				lenEntered, szEntry, IsUnicodeMode(), technology);
 
 	PRectangle rcClient = GetClientRectangle();
 	Point pt = LocationFromPosition(sel.MainCaret() - lenEntered);
